brayns/Brayns.cpp

```diff
@@ -389,11 +389,23 @@ struct Brayns::Impl : public PluginAPI
 
     void render()
     {
-        std::lock_guard<std::mutex> lock{_renderMutex};
+        {
+            std::lock_guard<std::mutex> lock{_renderMutex};
+
+            _renderTimer.start();
+            _engine->render();
+            _renderTimer.stop();
+        }
 
-        _renderTimer.start();
-        _engine->render();
-        _renderTimer.stop();
+        const auto& params = _parametersManager.getApplicationParameters();
+        const auto fps = params.getImageStreamFPS();
+        const auto delta = _renderTimer.perSecondSmoothed() - fps;
+        if (delta > 0)
+        {
+            const int64_t targetTime = (1. / fps) * 1000.f;
+            std::this_thread::sleep_for(std::chrono::milliseconds(
+                targetTime - _renderTimer.milliseconds()));
+        }
```

Comment:
This should either be called only when streaming is enabled, or the parameter should be renamed from imageStreamFPS to something more generic (with a way to adjust/disable it when benchmarking performance).

Reply:
I don't want to rename it now to not break the web API. It can already be changed on the commandline.

Reply:
OK, I would still rename it everywhere else, but up to you.

Reply:
I added a different parameter for this now. The tests were using the existing FPS to 'disable' streaming to speed up the execution of the tests. It also breaks nothing else at the moment hopefully.